Oops, got a lot of errors while clean rebuild:

```
make[5]: Entering directory '/home/krd/prj/xwrt/fork.openwrt/build_dir/target-aarch64_cortex-a53_glibc/xdp-tools-1.4.2/lib/util'
/home/krd/prj/xwrt/llvm-bpf-18.1.7.Linux-x86_64/bin/clang -S \
    -target mips64el-linux-gnu \
    -D __BPF_TRACING__ \
    -Wno-visibility -fno-stack-protector -nostdinc -isystem /home/krd/prj/xwrt/fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-13.3.0_glibc/usr/include -isystem /home/krd/prj/xwrt/fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-13.3.0_glibc/include -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/mach-generic -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/generated -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/include -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/uapi -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/generated/uapi -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/include/uapi -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/include/generated/uapi -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/tools/lib -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/tools/testing/selftests -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/samples/bpf -include linux/kconfig.h -include asm_goto_workaround.h -I/home/krd/prj/xwrt/fork.openwrt/build_dir/target-aarch64_cortex-a53_glibc/xdp-tools-1.4.2 -D__KERNEL__ -D__BPF_TRACING__ -DCONFIG_GENERIC_CSUM -D__TARGET_ARCH_mips -mlittle-endian -fno-stack-protector -Wall -Wno-unused-value -Wno-pointer-sign -Wno-compare-distinct-pointer-types -Wno-gnu-variable-sized-type-not-at-end -Wno-address-of-packed-member -Wno-tautological-compare -Wno-unknown-warning-option -fno-asynchronous-unwind-tables -Wno-uninitialized -Wno-unused-variable -Wno-unused-label -O2 -emit-llvm -Xclang -disable-llvm-passes -Wno-error -fno-stack-protector -DBPF_DIR_MNT=\"/sys/fs/bpf\" -DBPF_OBJECT_PATH=\"/usr/lib/bpf\" -DMAX_DISPATCHER_ACTIONS=10 -DTOOLS_VERSION=\""1.4.2"\" -DLIBBPF_VERSION=\"1.4.3\" -DRUNDIR=\"/tmp/run\" -DHAVE_BPFTOOL -DHAVE_LIBBPF_PERF_BUFFER__CONSUME -DHAVE_LIBBPF_BTF__LOAD_FROM_KERNEL_BY_ID -DHAVE_LIBBPF_BTF__TYPE_CNT -DHAVE_LIBBPF_BPF_OBJECT__NEXT_MAP -DHAVE_LIBBPF_BPF_OBJECT__NEXT_PROGRAM -DHAVE_LIBBPF_BPF_PROGRAM__INSN_CNT -DHAVE_LIBBPF_BPF_PROGRAM__TYPE -DHAVE_LIBBPF_BPF_PROGRAM__FLAGS -DHAVE_LIBBPF_BPF_PROGRAM__EXPECTED_ATTACH_TYPE -DHAVE_LIBBPF_BPF_MAP_CREATE -DHAVE_LIBBPF_PERF_BUFFER__NEW_RAW -DHAVE_LIBBPF_BPF_XDP_ATTACH -DHAVE_LIBBPF_BPF_MAP__SET_AUTOCREATE -DHAVE_LIBBPF_BPF_PROG_TEST_RUN_OPTS -DHAVE_LIBBPF_BPF_XDP_QUERY -DHAVE_SECURE_GETENV -DDEBUG -DLIBBPF_DYNAMIC -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64   -nostdinc -isystem /home/krd/prj/xwrt/fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-13.3.0_glibc/usr/include -isystem /home/krd/prj/xwrt/fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-13.3.0_glibc/include -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/mach-generic -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/generated -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/include -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/uapi -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/generated/uapi -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/include/uapi -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/include/generated/uapi -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/tools/lib -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/tools/testing/selftests -I/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/samples/bpf -include linux/kconfig.h -include asm_goto_workaround.h -I/home/krd/prj/xwrt/fork.openwrt/build_dir/target-aarch64_cortex-a53_glibc/xdp-tools-1.4.2 -D__KERNEL__ -D__BPF_TRACING__ -DCONFIG_GENERIC_CSUM -D__TARGET_ARCH_mips -mlittle-endian -fno-stack-protector -Wall -Wno-unused-value -Wno-pointer-sign -Wno-compare-distinct-pointer-types -Wno-gnu-variable-sized-type-not-at-end -Wno-address-of-packed-member -Wno-tautological-compare -Wno-unknown-warning-option -fno-asynchronous-unwind-tables -Wno-uninitialized -Wno-unused-variable -Wno-unused-label -O2 -emit-llvm -Xclang -disable-llvm-passes -Wno-error -fno-stack-protector -DBPF_DIR_MNT=\"/sys/fs/bpf\" -DBPF_OBJECT_PATH=\"/usr/lib/bpf\" -DMAX_DISPATCHER_ACTIONS=10 -DTOOLS_VERSION=\""1.4.2"\" -DLIBBPF_VERSION=\"1.4.3\" -DRUNDIR=\"/tmp/run\" -DHAVE_BPFTOOL -DHAVE_LIBBPF_PERF_BUFFER__CONSUME -DHAVE_LIBBPF_BTF__LOAD_FROM_KERNEL_BY_ID -DHAVE_LIBBPF_BTF__TYPE_CNT -DHAVE_LIBBPF_BPF_OBJECT__NEXT_MAP -DHAVE_LIBBPF_BPF_OBJECT__NEXT_PROGRAM -DHAVE_LIBBPF_BPF_PROGRAM__INSN_CNT -DHAVE_LIBBPF_BPF_PROGRAM__TYPE -DHAVE_LIBBPF_BPF_PROGRAM__FLAGS -DHAVE_LIBBPF_BPF_PROGRAM__EXPECTED_ATTACH_TYPE -DHAVE_LIBBPF_BPF_MAP_CREATE -DHAVE_LIBBPF_PERF_BUFFER__NEW_RAW -DHAVE_LIBBPF_BPF_XDP_ATTACH -DHAVE_LIBBPF_BPF_MAP__SET_AUTOCREATE -DHAVE_LIBBPF_BPF_PROG_TEST_RUN_OPTS -DHAVE_LIBBPF_BPF_XDP_QUERY -DHAVE_SECURE_GETENV -DDEBUG -DLIBBPF_DYNAMIC -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64   -I../../headers  \
    -Wall \
    -Wno-unused-value \
    -Wno-pointer-sign \
    -Wno-compare-distinct-pointer-types \
    -Werror \
    -O2 -emit-llvm -c -g -o xdp_sample.bpf.ll xdp_sample.bpf.c
In file included from xdp_sample.bpf.c:3:
../../headers/bpf/vmlinux.h:5:46: error: unknown attribute 'preserve_access_index' ignored [-Werror,-Wunknown-attributes]
    5 | #pragma clang attribute push (__attribute__((preserve_access_index)), apply_to = record)
      |                                              ^~~~~~~~~~~~~~~~~~~~~
../../headers/bpf/vmlinux.h:8:1: note: when applied to this declaration
    8 | struct net_device {
      | ^
../../headers/bpf/vmlinux.h:5:46: error: unknown attribute 'preserve_access_index' ignored [-Werror,-Wunknown-attributes]
    5 | #pragma clang attribute push (__attribute__((preserve_access_index)), apply_to = record)
      |                                              ^~~~~~~~~~~~~~~~~~~~~
../../headers/bpf/vmlinux.h:12:1: note: when applied to this declaration
   12 | struct xdp_cpumap_stats {
      | ^
../../headers/bpf/vmlinux.h:5:46: error: unknown attribute 'preserve_access_index' ignored [-Werror,-Wunknown-attributes]
    5 | #pragma clang attribute push (__attribute__((preserve_access_index)), apply_to = record)
      |                                              ^~~~~~~~~~~~~~~~~~~~~
../../headers/bpf/vmlinux.h:18:1: note: when applied to this declaration
   18 | struct bpf_prog {
      | ^
../../headers/bpf/vmlinux.h:5:46: error: unknown attribute 'preserve_access_index' ignored [-Werror,-Wunknown-attributes]
    5 | #pragma clang attribute push (__attribute__((preserve_access_index)), apply_to = record)
      |                                              ^~~~~~~~~~~~~~~~~~~~~
../../headers/bpf/vmlinux.h:21:1: note: when applied to this declaration
   21 | struct bpf_map {
      | ^
In file included from xdp_sample.bpf.c:4:
In file included from /home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/include/linux/bpf.h:9:
In file included from /home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/include/linux/workqueue.h:9:
In file included from /home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/include/linux/timer.h:5:
In file included from /home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/include/linux/list.h:9:
In file included from /home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/include/linux/kernel.h:12:
In file included from /home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/include/linux/bitops.h:33:
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:101:3: error: unused variable '__temp' [-Werror,-Wunused-variable]
  101 |                 __bit_op(*m, __INS "%0, %3, %2, 1", "i"(bit), "r"(~0));
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:29:16: note: expanded from macro '__bit_op'
   29 |         unsigned long __temp;                                   \
      |                       ^~~~~~
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:105:2: error: unused variable '__temp' [-Werror,-Wunused-variable]
  105 |         __bit_op(*m, "or\t%0, %2", "ir"(BIT(bit)));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:29:16: note: expanded from macro '__bit_op'
   29 |         unsigned long __temp;                                   \
      |                       ^~~~~~
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:92:26: error: unused variable 'm' [-Werror,-Wunused-variable]
   92 |         volatile unsigned long *m = &addr[BIT_WORD(nr)];
      |                                 ^
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:129:3: error: unused variable '__temp' [-Werror,-Wunused-variable]
  129 |                 __bit_op(*m, __INS "%0, $0, %2, 1", "i"(bit));
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:29:16: note: expanded from macro '__bit_op'
   29 |         unsigned long __temp;                                   \
      |                       ^~~~~~
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:133:2: error: unused variable '__temp' [-Werror,-Wunused-variable]
  133 |         __bit_op(*m, "and\t%0, %2", "ir"(~BIT(bit)));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:29:16: note: expanded from macro '__bit_op'
   29 |         unsigned long __temp;                                   \
      |                       ^~~~~~
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:120:26: error: unused variable 'm' [-Werror,-Wunused-variable]
  120 |         volatile unsigned long *m = &addr[BIT_WORD(nr)];
      |                                 ^
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:169:2: error: unused variable '__temp' [-Werror,-Wunused-variable]
  169 |         __bit_op(*m, "xor\t%0, %2", "ir"(BIT(bit)));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:29:16: note: expanded from macro '__bit_op'
   29 |         unsigned long __temp;                                   \
      |                       ^~~~~~
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:161:26: error: unused variable 'm' [-Werror,-Wunused-variable]
  161 |         volatile unsigned long *m = &addr[BIT_WORD(nr)];
      |                                 ^
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:162:6: error: unused variable 'bit' [-Werror,-Wunused-variable]
  162 |         int bit = nr % BITS_PER_LONG;
      |             ^~~
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:190:10: error: unused variable '__temp' [-Werror,-Wunused-variable]
  190 |                 orig = __test_bit_op(*m, "%0",
      |                        ^~~~~~~~~~~~~~~~~~~~~~~
  191 |                                      "or\t%1, %0, %3",
      |                                      ~~~~~~~~~~~~~~~~~
  192 |                                      "ir"(BIT(bit)));
      |                                      ~~~~~~~~~~~~~~~
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:46:24: note: expanded from macro '__test_bit_op'
   46 |         unsigned long __orig, __temp;                           \
      |                               ^~~~~~
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:183:26: error: unused variable 'm' [-Werror,-Wunused-variable]
  183 |         volatile unsigned long *m = &addr[BIT_WORD(nr)];
      |                                 ^
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:190:10: error: variable '__orig' is uninitialized when used here [-Werror,-Wuninitialized]
  190 |                 orig = __test_bit_op(*m, "%0",
      |                        ^~~~~~~~~~~~~~~~~~~~~~~
  191 |                                      "or\t%1, %0, %3",
      |                                      ~~~~~~~~~~~~~~~~~
  192 |                                      "ir"(BIT(bit)));
      |                                      ~~~~~~~~~~~~~~~
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:62:2: note: expanded from macro '__test_bit_op'
   62 |         __orig;                                                 \
      |         ^~~~~~
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:190:10: note: variable '__orig' is declared here
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:46:2: note: expanded from macro '__test_bit_op'
   46 |         unsigned long __orig, __temp;                           \
      |         ^
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:236:9: error: unused variable '__temp' [-Werror,-Wunused-variable]
  236 |                 res = __test_bit_op(*m, "%1",
      |                       ^~~~~~~~~~~~~~~~~~~~~~~
  237 |                                     __EXT "%0, %1, %3, 1;"
      |                                     ~~~~~~~~~~~~~~~~~~~~~~
  238 |                                     __INS "%1, $0, %3, 1",
      |                                     ~~~~~~~~~~~~~~~~~~~~~~
  239 |                                     "i"(bit));
      |                                     ~~~~~~~~~
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:46:24: note: expanded from macro '__test_bit_op'
   46 |         unsigned long __orig, __temp;                           \
      |                               ^~~~~~
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:241:10: error: unused variable '__temp' [-Werror,-Wunused-variable]
  241 |                 orig = __test_bit_op(*m, "%0",
      |                        ^~~~~~~~~~~~~~~~~~~~~~~
  242 |                                      "or\t%1, %0, %3;"
      |                                      ~~~~~~~~~~~~~~~~~
  243 |                                      "xor\t%1, %1, %3",
      |                                      ~~~~~~~~~~~~~~~~~~
  244 |                                      "ir"(BIT(bit)));
      |                                      ~~~~~~~~~~~~~~~
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:46:24: note: expanded from macro '__test_bit_op'
   46 |         unsigned long __orig, __temp;                           \
      |                               ^~~~~~
/home/krd/prj/xwrt/fork.openwrt/staging_dir/target-aarch64_cortex-a53_glibc/bpf-headers/arch/mips/include/asm/bitops.h:227:26: error: unused variable 'm' [-Werror,-Wunused-variable]
  227 |         volatile unsigned long *m = &addr[BIT_WORD(nr)];
      |                                 ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
```

Config:
```sh
$ grep -Fi -e bpf -e btf -e llvm -e xdp .config
CONFIG_KERNEL_DEBUG_INFO_BTF=y
CONFIG_KERNEL_DEBUG_INFO_BTF_MODULES=y
# CONFIG_KERNEL_MODULE_ALLOW_BTF_MISMATCH is not set
CONFIG_KERNEL_BPF_EVENTS=y
# CONFIG_KERNEL_BPF_KPROBE_OVERRIDE is not set
# CONFIG_KERNEL_LWTUNNEL_BPF is not set
CONFIG_KERNEL_XDP_SOCKETS=y
# CONFIG_BPF_TOOLCHAIN_NONE is not set
CONFIG_BPF_TOOLCHAIN_HOST=y
# CONFIG_BPF_TOOLCHAIN_BUILD_LLVM is not set
CONFIG_BPF_TOOLCHAIN_HOST_PATH="/home/krd/prj/xwrt/llvm-bpf-18.1.7.Linux-x86_64"
CONFIG_HAS_BPF_TOOLCHAIN=y
# CONFIG_HAS_PREBUILT_LLVM_TOOLCHAIN is not set
CONFIG_USE_LLVM_HOST=y
# CONFIG_PACKAGE_kselftests-bpf is not set
# CONFIG_PACKAGE_kmod-bpf-test is not set
CONFIG_PACKAGE_kmod-sched-bpf=y
CONFIG_PACKAGE_kmod-xdp-sockets-diag=y
CONFIG_PACKAGE_libbpf=y
# CONFIG_PACKAGE_libpfring is not set
CONFIG_PACKAGE_libxdp=y
CONFIG_PACKAGE_tc-bpf=y
# CONFIG_PACKAGE_bpfcountd is not set
CONFIG_PACKAGE_bpftool-full=y
# CONFIG_PACKAGE_bpftool-minimal is not set
CONFIG_PACKAGE_xdp-filter=y
CONFIG_PACKAGE_xdp-loader=y
CONFIG_PACKAGE_xdpdump=y
CONFIG_PACKAGE_ucode-mod-bpf=y
```

--- a/package/network/utils/xdp-tools/Makefile
+++ b/package/network/utils/xdp-tools/Makefile
@@ -2,8 +2,8 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=xdp-tools
 PKG_RELEASE:=1
-PKG_VERSION:=1.2.9
-PKG_HASH:=159ed8d3c8195d812ec3cde83bd736245a72743af372998320d39c2ba69ab142
+PKG_VERSION:=1.4.2
+PKG_HASH:=49c2b96b2be878449a797a74ca515a63e13418cb8ea904df08f8ef9cb2ac5570
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://codeload.github.com/xdp-project/xdp-tools/tar.gz/v$(PKG_VERSION)?

--- a/package/network/utils/xdp-tools/patches/010-configure-respect-LDFLAGS.patch
+++ b/package/network/utils/xdp-tools/patches/010-configure-respect-LDFLAGS.patch
@@ -1,6 +1,6 @@
 --- a/configure
 +++ b/configure
-@@ -174,7 +174,7 @@ int main(int argc, char **argv) {
+@@ -191,7 +191,7 @@ int main(int argc, char **argv) {
      return 0;
  }
  EOF
@@ -9,16 +9,16 @@
      if [ "$?" -eq "0" ]; then
          echo "HAVE_PCAP:=y" >>$CONFIG
          [ -n "$LIBPCAP_CFLAGS" ] && echo 'CFLAGS += ' $LIBPCAP_CFLAGS >> $CONFIG
-@@ -222,7 +222,7 @@ int main(int argc, char **argv) {
+@@ -267,7 +267,7 @@ int main(int argc, char **argv) {
      return 0;
  }
  EOF
--    libbpf_err=$($CC -o $TMPDIR/libbpftest $TMPDIR/libbpftest.c -Werror $LIBBPF_CFLAGS $LIBBPF_LDLIBS 2>&1)
-+    libbpf_err=$($CC -o $TMPDIR/libbpftest $TMPDIR/libbpftest.c -Werror $LIBBPF_CFLAGS $LIBBPF_LDLIBS $LDFLAGS 2>&1)
+-    compile_cmd="$CC -o $TMPDIR/libbpftest $TMPDIR/libbpftest.c -Werror $LIBBPF_CFLAGS $LIBBPF_LDLIBS"
++    compile_cmd="$CC -o $TMPDIR/libbpftest $TMPDIR/libbpftest.c -Werror $LIBBPF_CFLAGS $LIBBPF_LDLIBS $LDFLAGS"
+     libbpf_err=$($compile_cmd 2>&1)
      if [ "$?" -eq "0" ]; then
          echo "HAVE_FEATURES+=${config_var}" >>"$CONFIG"
-         echo "yes"
-@@ -289,7 +289,7 @@ int main(int argc, char **argv) {
+@@ -345,7 +345,7 @@ int main(int argc, char **argv) {
  }
  EOF
  

--- a/package/network/utils/xdp-tools/patches/020-libxdp-Use-__noinline__-reserved-attribute-for-XDP-d.patch
+++ b/package/network/utils/xdp-tools/patches/020-libxdp-Use-__noinline__-reserved-attribute-for-XDP-d.patch
@@ -18,7 +18,7 @@ Signed-off-by: Andre Heider <a.heider@gmail.com>
 
 --- a/lib/libxdp/protocol.org
 +++ b/lib/libxdp/protocol.org
-@@ -54,7 +54,7 @@ static volatile const struct xdp_dispatc
+@@ -59,7 +59,7 @@ static volatile const struct xdp_dispatc
  /* The volatile return value prevents the compiler from assuming it knows the
   * return value and optimising based on that.
   */
@@ -29,7 +29,7 @@ Signed-off-by: Andre Heider <a.heider@gmail.com>
  
 --- a/lib/libxdp/xdp-dispatcher.c.in
 +++ b/lib/libxdp/xdp-dispatcher.c.in
-@@ -30,7 +30,7 @@ static volatile const struct xdp_dispatc
+@@ -29,7 +29,7 @@ static volatile const struct xdp_dispatc
   * return value and optimising based on that.
   */
  forloop(`i', `0', NUM_PROGS,
@@ -38,7 +38,7 @@ Signed-off-by: Andre Heider <a.heider@gmail.com>
  int format(`prog%d', i)(struct xdp_md *ctx) {
          volatile int ret = XDP_DISPATCHER_RETVAL;
  
-@@ -40,7 +40,7 @@ int format(`prog%d', i)(struct xdp_md *c
+@@ -39,7 +39,7 @@ int format(`prog%d', i)(struct xdp_md *c
  }
  ')
  

--- a/package/network/utils/xdp-tools/patches/023-libxdp-fix-compilation-on-multiarch-systems.patch
+++ b/package/network/utils/xdp-tools/patches/023-libxdp-fix-compilation-on-multiarch-systems.patch
@@ -19,7 +19,7 @@ Signed-off-by: Andre Heider <a.heider@gmail.com>
 
 --- a/lib/libxdp/Makefile
 +++ b/lib/libxdp/Makefile
-@@ -30,7 +30,7 @@ PC_FILE := $(OBJDIR)/libxdp.pc
+@@ -28,7 +28,7 @@ PC_FILE := $(OBJDIR)/libxdp.pc
  TEMPLATED_SOURCES := xdp-dispatcher.c
  
  CFLAGS += -I$(HEADER_DIR)

--- a/package/network/utils/xdp-tools/patches/024-lib-allow-overwriting-W-flags-via-BPF_CFLAGS.patch
+++ b/package/network/utils/xdp-tools/patches/024-lib-allow-overwriting-W-flags-via-BPF_CFLAGS.patch
@@ -17,7 +17,7 @@ Signed-off-by: Andre Heider <a.heider@gmail.com>
 
 --- a/lib/common.mk
 +++ b/lib/common.mk
-@@ -108,12 +108,12 @@ $(XDP_OBJ): %.o: %.c $(KERN_USER_H) $(EX
+@@ -111,12 +111,12 @@ $(XDP_OBJ): %.o: %.c $(KERN_USER_H) $(EX
  	$(QUIET_CLANG)$(CLANG) -S \
  	    -target $(BPF_TARGET) \
  	    -D __BPF_TRACING__ \
@@ -33,7 +33,7 @@ Signed-off-by: Andre Heider <a.heider@gmail.com>
  
 --- a/lib/libxdp/Makefile
 +++ b/lib/libxdp/Makefile
-@@ -139,12 +139,12 @@ $(XDP_OBJS): %.o: %.c $(BPF_HEADERS) $(L
+@@ -138,12 +138,12 @@ $(XDP_OBJS): %.o: %.c $(BPF_HEADERS) $(L
  	$(QUIET_CLANG)$(CLANG) -S \
  	    -target $(BPF_TARGET) \
  	    -D __BPF_TRACING__ \

--- a/package/network/utils/xdp-tools/patches/025-Add-BPF_LDFLAGS-to-allow-overwriting-llc-s-march-arg.patch
+++ b/package/network/utils/xdp-tools/patches/025-Add-BPF_LDFLAGS-to-allow-overwriting-llc-s-march-arg.patch
@@ -33,18 +33,18 @@ Signed-off-by: Andre Heider <a.heider@gmail.com>
  find_tool()
 --- a/lib/common.mk
 +++ b/lib/common.mk
-@@ -115,7 +115,7 @@ $(XDP_OBJ): %.o: %.c $(KERN_USER_H) $(EX
+@@ -118,7 +118,7 @@ $(XDP_OBJ): %.o: %.c $(KERN_USER_H) $(EX
  	    -Werror \
  	    $(BPF_CFLAGS) \
  	    -O2 -emit-llvm -c -g -o ${@:.o=.ll} $<
 -	$(QUIET_LLC)$(LLC) -march=$(BPF_TARGET) -filetype=obj -o $@ ${@:.o=.ll}
 +	$(QUIET_LLC)$(LLC) $(BPF_LDFLAGS) -filetype=obj -o $@ ${@:.o=.ll}
  
- .PHONY: man
- ifeq ($(EMACS),)
+ $(BPF_SKEL_H): %.skel.h: %.bpf.o
+ 	$(QUIET_GEN)$(BPFTOOL) gen skeleton $< name ${@:.skel.h=} > $@
 --- a/lib/libxdp/Makefile
 +++ b/lib/libxdp/Makefile
-@@ -146,7 +146,7 @@ $(XDP_OBJS): %.o: %.c $(BPF_HEADERS) $(L
+@@ -145,7 +145,7 @@ $(XDP_OBJS): %.o: %.c $(BPF_HEADERS) $(L
  	    -Werror \
  	    $(BPF_CFLAGS) \
  	    -O2 -emit-llvm -c -g -o ${@:.o=.ll} $<

